### ISSUE TITLE
[dev] get better console.log() messages

### ISSUE THREAD

**Issue by PierBover** [cloudflare/wrangler#1249]
### 💡 Feature request

It would be great if `dev` provided better `console.log()` messages.

#### Overview and problem statement

When using `console.log()` objects get truncated. Not only nested objects and array but look at the ellipsis at the far right:

![image](https://user-images.githubusercontent.com/226483/81212896-3596ce80-8f9b-11ea-83cc-e1e85c29ab5b.png)

A common trick in Node is to use `JSON.stringify` to get the full log, but when doing this line breaks and white space are printed quite literally...

![image](https://user-images.githubusercontent.com/226483/81213105-83abd200-8f9b-11ea-9875-93fabac71b03.png)

The best solution would be to print complete objects and nested objects n levels deep, but if that is not feasible at least interpret strings correctly.

**Comment by EverlastingBugstopper** [comment]
Yeah! This is actually a really interesting problem. The way `wrangler dev` works is that it connects to a WebSocket that emits log events with the Chrome Devtools Protocol. The problem with that is that the protocol itself is very interactive, and not linear. The usual way that you interact with it is with the Chrome Devtools Console, if you `conosle.log(myLargeObject)` it will show you the fields with a little dropdown. You can click the dropdown and the front end will send a websocket message asking for the next nested level of the object, and then when it gets it back it will display it.

Since `wrangler dev` is run in the terminal, that is pretty much impossible (though I am open to ideas!) My thought here is to solve this use case with #946.

As for the second request - printing line breaks and white spaces - that's definitely something we should look into making better. The repo we use to serialize/deserialize these messages is [here](https://github.com/EverlastingBugstopper/chrome-devtools-rs)

**Comment by EverlastingBugstopper** [comment]
ok - just to clarify, the solution to this issue is not #946 and we should print out the `JSON.stringify` stuff in a much cleaner manner

**Comment by PierBover** [comment]
I see there is a PR that supposedly fixed this, but I'm on `wrangler 1.15.0` and still getting these broken `console.log` messages:

![image](https://user-images.githubusercontent.com/226483/115488238-c94ec280-a21f-11eb-82dc-4e329e91bc8e.png)

**Comment by jspspike** [comment]
Hey @PierBover I can try and take a look at why you're not getting formatted json. What exactly are doing to print the request headers? When I attempt
```javascript
let requestHeaders = JSON.stringify(Object.fromEntries(request.headers), null, 2);
console.log(`Request headers: ${requestHeaders}`);
```
from the example you linked I get
```
Request headers: {
  "accept": "*/*",
  "accept-encoding": "gzip",
  "cf-connecting-ip": "72.182.114.2",
  "cf-ew-preview-server": "https://15m372.cfops.it",
  "cf-ipcountry": "US",
  "cf-ray": "66e4a627ba760bd9",
  "cf-visitor": "{\"scheme\":\"https\"}",
  "cf-workers-preview-token": "Aa6eZgsxPxau0p576nJ5AkvdKHpIn_ildawHk8qgC71jEdlOFJi5uN7HXoixOaISMStQxBpyIyTEcKUfbXp7yfoXKFSzF2BjysoINAnuezkpBPMgxyXOs8mPDu9K4QHhn_kMNmQWBHAOirlyInTryIfV4DNGSUnyGRTsUtiUdMfSB8jpCzhwdD-fCe3LnF8Z9syHTePjPtDjkBU96RFfv2zTTKpiOfI6HaVtct6mPuhyTpfitJHCuwzI6HJD_oKlCTFKEt9HiwwAWzrxovn1PuTjUw2afia7MVbbFTl5RmWSKf4Jexez9lVtpEzQXxrOGDUq7RdG4GUolg6qsJkgS4Kq4th3ZITewsa-C8cvYq2RYMT3OmEGGYGpcGh6B3j3_BvIQl2-58IO0QokV3lm7MsekfCzzMmU4STgcS-3n_4x9JLvCIp06-F1e24wqG-98p_QV-VXoXE4xWmkm5hqY33F5ACU7hZNOMPX-bSgIDNAfD1Av3HClaN9DG4qkUQx95mkwErDOrGqikoXJAakVyhiRO7zKz5pHzZ4TkyYd0YFZH1QCHp88nGIlIj9dTGRy9luB_75HmTBCK4OejqzddptLLsXhusfWPycZcq6S3vT8Q1wZ4VBwCY8KGrg4Vez5mWHplBqWXH3TodRB3TilQ2ItJl6EJcf6SvBIQNf8ql9v_hGFCmP9YRP_V02gXDAcr5sXM2Vx0QkWz67RbX94UPL3-4MpYpTOKTSF2dzwNlvOpXWdGoByaV2oNRhjPQABAkQCamQ9STgRAz6qfvasbziWHKMV3d2asG9_Vwc68pF",
  "connection": "Keep-Alive",
  "host": "worker.ftc.workers.dev",
  "user-agent": "curl/7.77.0",
  "x-forwarded-proto": "https",
  "x-real-ip": "72.182.114.2"
}
```
